lib/elixir/lib/exception.ex

```diff
@@ -765,6 +765,24 @@ defmodule UnicodeConversionError do
   end
 end
 
+defmodule StartLinkNameError do
+  defexception [arg: nil]
+
+  def message(exception) do
+    """
+    invalid argument type given for `name`.
+
+    Valid types are:
+      * nil
+      * atom
+      * tuple
```

Comment:
Yes, if we are going to give helpful warning we should probably handle "bad" tuples too, otherwise that will lead to a function clause or case clause deep inside OTP :smile: